### DISCUSSION TITLE
Error when Changing DENO_DIR

### DISCUSSION THREAD

**Discussion by wwderw** [mandarineorg/leaf#7] [Q&A]
In order to not have an issue for others and the cache always being cleared out, I usually change the cache directory to something that is within the file tree of the program (and thus gets distributed with the binary), so it doesn't go to the Deno default location.

However, when I compile with leaf after changing the DENO_DIR location and I run the resulting binary, it tells me that I need the --allow-env flag (which I ran when I did deno run to compile with leaf).  Is there something that I need to do within the js file that I use to compile with leaf or something else that I need to be doing?

**Comment by shadowtime2000** [comment]
@wwderw Make sure when you use `Leaf.compile` you pass in `--allow-env` in the flags, because you are allowing the process which compiles the binary to use the env but you need to allow the binary itself with Deno to use the env.

**Comment by wwderw** [comment]
I did when I did deno run, is there another time that I need to?

**Comment by shadowtime2000** [comment]
@wwderw Yes, in the calling of `Leaf.compile` in the options set `flags` to `["--allow-env"]` and it should work.

**Comment by wwderw** [comment]
That did it.  Thanks!
